astroplan/scheduling.py

```diff
@@ -90,6 +90,80 @@ def from_exposures(cls, target, priority, time_per_exposure,
         return ob
 
 
+class Scorer(object):
+    """
+    Returns scores and score arrays from the evaluation of constraints on
+    observing blocks
+    """
+    def __init__(self, blocks, observer, schedule):
+        """
+
+        Parameters
+        ----------
+        blocks : list of `~astroplan.scheduling.ObservingBlock` objects
+            list of blocks that need to be scored
+        observer : `~astroplan.Observer`
+            the observer
+        schedule : `~astroplan.scheduling.Schedule`
+            The schedule inside which the blocks should fit
+        """
+        self.blocks = blocks
+        self.observer = observer
+        self.schedule = schedule
+        # the lists below should change if there are new constraints made
+        self.constraint_list = ['AltitudeConstraint', 'AirmassConstraint', 'AtNightConstraint',
```

Comment:
`AirmassConstraint` subclassing `AltitudeConstriant` is useful in other places: the schedule needs to make sure there is some `Constraint` on the altitude, so it can just check if any of the constraints are an `AltitudeConstraint`. The MetaClass idea seems interesting, but I'm not sure it's necessary. A recursive version of `__subclasses__()` should be able to handle it (and any user-created `Constraints`).